### PR TITLE
Bug 1124112 - [Stingray] Seamlessly launch search app from home

### PR DESCRIPTION
**Pull request by luke-chang**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c25c3a143273)

File: tv_apps/smart-system/js/app_window_manager.js [hunk 1]
```diff
@@ -222,6 +222,10 @@
           immediateTranstion = true;
         }
 
+        if (appNext.manifest && appNext.manifest.role == 'search') {
```
Comment (evelynhung) [inline tv_apps/smart-system/js/app_window_manager.js]: === ?

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=faa758c73f9d)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=a57048f3d125)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7b61875076ce)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=fc067fff4b19)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=436e02a20793)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e7c9eefa054d)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=0a33135e8dbf)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e7bc0e7faf39)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7a53c74e537b)

**Comment by try-server-hook** [comment]
[luke-chang Luke Chang (luke-chang)](https://github.com/luke-chang) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c39737acf0e0)

File: tv_apps/smart-home/js/home.js [hunk 2]
```diff
@@ -322,12 +322,18 @@
     },
 
     onSearchBarShown: function() {
-      /*jshint nonew: false */
-      new MozActivity({
+      var hideSearchBar = function() {
+        document.removeEventListener('visibilitychange', hideSearchBar);
+        this.searchBar.hide();
+      }.bind(this);
+      document.addEventListener('visibilitychange', hideSearchBar);
```
Comment (huchengtw-moz) [inline tv_apps/smart-home/js/home.js]: How about this one:

document.addEventListener('visibilitychange', function hideSearchBar() {
  document.removeEventListener('visibilitychange', hideSearchBar);
  this.searchBar.hide();
}.bind(this));
```

Comment (luke-chang) [inline tv_apps/smart-home/js/home.js]: If so, the "removeEventListener" should be unable to get the instance of the "binded hideSearchBar".

**Pull request merged**